frontend/packages/dev-console/src/components/topology/filters/TopologyFilterBar.tsx

```diff
@@ -71,15 +73,53 @@ const TopologyFilterBar: React.FC<TopologyFilterBarProps> = ({
   const kialiLink = getNamespaceDashboardKialiLink(consoleLinks, namespace);
   const queryParams = useQueryParams();
   const searchQuery = queryParams.get('searchQuery') || '';
+  const [isQuickSearchActive, setIsQuickSearchActive] = React.useState<boolean>(false);
 
   const onTextFilterChange = (text) => {
     const query = text?.trim();
     onSearchChange(query);
   };
 
+  React.useEffect(() => {
+    const onKeyDown = (e: KeyboardEvent) => {
+      e.preventDefault();
+      const { nodeName } = e.target as Element;
+      if (nodeName === 'INPUT' || nodeName === 'TEXTAREA') {
+        return;
+      }
+      switch (e.code) {
+        case 'KeyF':
+          e.ctrlKey && setIsQuickSearchActive(true);
+          break;
+        case 'Escape':
+          setIsQuickSearchActive(false);
+          break;
+        default:
+          break;
+      }
+    };
+
+    window.addEventListener('keydown', onKeyDown);
+
+    return () => {
+      window.removeEventListener('keydown', onKeyDown);
+    };
+  });
```

Comment:
Use effect without an empty dependency array will run on every render. You want to use dependency of `[]`

Reply:
Yes, missed it. :P